thunder/core/transforms.py

```diff
@@ -3768,9 +3769,91 @@ def backward_fn(saved_for_backward, cotangents):
     _update_backward_with_new_saved_for_backward(backward_trace, only_used_bw_saved_for_backward)
     forward_trace.set_provenance(TraceProvenance("Augmented forward pass"))
     backward_trace.set_provenance(TraceProvenance("Backward pass"))
+
+    disable_saved_for_backward_recomputation: None | bool = get_compile_option(
+        "disable_saved_for_backward_recomputation", "Disable save for backward tensors recomputation."
+    )
+    if not disable_saved_for_backward_recomputation:
+        forward_trace, backward_trace = recompute_saved_for_backward(forward_trace, backward_trace)
+
     return ForwardBackwardTraces(forward_trace, backward_trace)
 
 
+def recompute_saved_for_backward(fw_trace: Trace, bw_trace: Trace) -> tuple[Trace, Trace]:
+    """Generates the pair of traces with rematerializaion of the saved-for-backward tensors.
+    Args:
+        fw_trace (Trace): forward trace where to get the saved for backward from.
+        bw_trace (Trace): backward trace where to recompute the saved for backward to.
+
+    Returns:
+        tuple[Trace, Trace]: A tuple containing the new forward and backward traces.
+    """
+    start_time_ns = time.perf_counter_ns()
+    # First of all we need to get the saved for backward tensors
+    saved_for_bw = get_saved_for_backward_tensors(fw_trace)
+
+    # Let's create these two maps that will help writing the logic
+    fw_trace_args = {j.name: j for j in fw_trace.args}
```

Comment:
Thanks for this suggestion! Initially i tought that it would just become too ugly to use this interface all over but it turns out it can help create sets and it doesn't look that bad. Sets allow to do the operations we need to exclude/include symbols very easily